### DISCUSSION TITLE
Hybrid Rendering

### DISCUSSION THREAD

**Discussion by Abiti-936** [solidjs/solid-start#252] [General]
First of all, I need to say, You are doing an Awesome Job. I really like how Solid Start Ships Minimal JS, can Render at any hosting platform, how it support Streaming Rendering & the Experimental Island Mode are just Really Great.

Now, what I would like to ask is about switching Rendering Modes per Route Basis? I don't use Next.js, but I like their approach about these problem & I think everything can't be Static generated, Stream Rended or Async Rended or can be Island based. So Are there any plans to Support Hybrid Rendering in the Future? Thanks.

**Comment by ryansolid** [comment]
It's possible. However it is more complicated than it might seem because of architectural differences. When  Next app creates a static route I believe you can still client navigate to it if you landed on a different page. This is something we will definitely do. However, the same is not true with MPA or Hybrid approaches because they are different in data requirements and how they handle shared state and how they are bundled. A Island page literally could not be rendered in the browser without having completely different bundled code. I think there is possibility here in terms of the client router being the bridge to client and server-rendered pages but it would place certain restrictions on those pages. To the point I'd argue that it is in itself a new "Hybrid" architecture.

We want to explore what that looks like but I suspect it will be a little different than opting in to modes per page. But rather that is the mode. So very much on our mind but still something that needs to be flushed out.
